DXMainClient/DXGUI/Multiplayer/PlayerListBox.cs

```diff
@@ -0,0 +1,178 @@
+using DTAClient.Online;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Rampastring.XNAUI;
+using Rampastring.XNAUI.XNAControls;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace DTAClient.DXGUI.Multiplayer
+{
+    public class PlayerListBox: XNAListBox
+    {
+        public List<ChannelUser> Users;
+
+        private Texture2D adminGameIcon;
+        private Texture2D unknownGameIcon;
+        private Texture2D badgeGameIcon;
+        private Texture2D friendIcon;
+
+        private XNAScrollBar scrollBar;
+
+        private GameCollection gameCollection;
+
+        public PlayerListBox(WindowManager windowManager, GameCollection gameCollection) : base(windowManager)
+        {
+            this.gameCollection = gameCollection;
+            
+            Users = new List<ChannelUser>();
+
+            adminGameIcon = AssetLoader.TextureFromImage(ClientCore.Properties.Resources.cncneticon);
+            unknownGameIcon = AssetLoader.TextureFromImage(ClientCore.Properties.Resources.unknownicon);
+            friendIcon = AssetLoader.LoadTexture("friendicon.png");
+            badgeGameIcon = AssetLoader.LoadTexture("Badges\\badge.png");
+
+            scrollBar = new XNAScrollBar(WindowManager);
+        }
+
+        public void AddUser(ChannelUser user)
+        {
+            AddUserToList(user);
+        }
+
+        /// <summary>
+        /// Refreshes game information in the game list box.
+        /// </summary>
+        public void Refresh()
+        {
+            Items.Clear();
+        }
+
+        public override void Draw(GameTime gameTime)
+        {
+            Rectangle windowRectangle = WindowRectangle();
+            
+            DrawPanel();
+
+            int height = 2;
+            int margin = 2;
+
+            for (int i = TopIndex; i < Items.Count; i++)
+            {
+                XNAListBoxItem lbItem = Items[i];
+                var user = (ChannelUser)lbItem.Tag;
+
+                if (height + lbItem.TextLines.Count * LineHeight > ClientRectangle.Height)
+                    break;
+
+                int x = TextBorderDistance;
+
+                if (i == SelectedIndex)
+                {
+                    int drawnWidth;
+
+                    if (DrawSelectionUnderScrollbar || !scrollBar.IsDrawn() || !EnableScrollbar)
+                    {
+                        drawnWidth = windowRectangle.Width - 2;
+                    }
+                    else
+                    {
+                        drawnWidth = windowRectangle.Width - 2 - scrollBar.ClientRectangle.Width;
+                    }
+
+                    Renderer.FillRectangle(
+                        new Rectangle(windowRectangle.X + 1, windowRectangle.Y + height,
+                        drawnWidth, lbItem.TextLines.Count * LineHeight),
+                        GetColorWithAlpha(FocusColor));
+                }
+
+                if (user.IsAdmin)
```

Comment:
Wouldn't it be possible to just draw lbItem.Texture here using Renderer.DrawTexture instead of performing these checks? The checks in AddUserToList look identical.